### DISCUSSION TITLE
Make Code component not crash entire page when providing lanugage that is not supported

### DISCUSSION THREAD

**Discussion by Szymon-Wrzos** [withastro/roadmap#781] [Proposal]
### Body

# Summary

I want Code component to not crash entire page when provided language is not in supported range 

# Background & Motivation
I was creating site with sanity cms and I had code component that can provide variety of langauges that may be not supported by Astro's `<Code />` component causing entire page to crash which is an issue especially during SSR

# Suggestion
Add fallback language prop that'll be used if provided language is not supported

# Example
```astro
---
import { Code } from "astro:components";
---
<Code
   code="...code..."
   lang="groq" // groq is definitely not supported by Shiki
   fallbackLang="txt" 
/>

```
# Current workaround
Currently, to mitigate this issues, it's necessary to actualy install shiki, get all supported languages and filter them out against provided from external source

In case of Astro migrating to another syntax highlighter, it'd be an issue since it'd require migrations from coder's site

**Comment by florian-lefebvre** [comment]
Do you think always falling back to text (directly in Astro, not through prop) would be enough?

**Reply by Szymon-Wrzos** [reply]
To think of it, yeah, you're right

**Reply by florian-lefebvre** [reply]
I'll give it a shot this week-end or start of next week. If I forget about it (ie. I don't ping you in a PR), feel free to ping me here!
